src/openapi/streaming/streaming.js

```diff
@@ -523,6 +525,103 @@ Streaming.prototype.disposeSubscription = function(subscription) {
     }
 };
 
+function handleSubscriptionReadyForUnsubscribe(subscriptions, resolve) {
+    let allSubscriptionsReady = true;
+    for (let i = 0; i < subscriptions.length && allSubscriptionsReady; i++) {
+        if (!subscriptions[i].isReadyForUnsubscribeByTag()) {
+            allSubscriptionsReady = false;
+        }
+    }
+
+    if (allSubscriptionsReady) {
+        resolve();
+    }
+}
+
+function getSubscriptionsToRemove(serviceGroup, url, tag) {
+    const subscriptionsToRemove = [];
+
+    for (let i = 0; i < this.subscriptions.length; i++) {
+        const subscription = this.subscriptions[i];
+
+        if (subscription.serviceGroup === serviceGroup &&
+                subscription.url === url &&
+                subscription.subscriptionData.Tag === tag) {
+
+            subscriptionsToRemove.push(subscription);
+        }
+    }
+
+    return subscriptionsToRemove;
+}
+
+function unsubscribeSubscriptionByTag(serviceGroup, url, tag, shouldDisposeSubscription) {
+    const subscriptionsToRemove = getSubscriptionsToRemove.call(this, serviceGroup, url, tag);
+
+    let allSubscriptionsReadyResolve;
+    const allSubscriptionsReady = new Promise((resolve) => {
+        allSubscriptionsReadyResolve = resolve;
+    });
+    const onStateChanged = handleSubscriptionReadyForUnsubscribe.bind(this, subscriptionsToRemove, allSubscriptionsReadyResolve);
+
+    for (let i = 0; i < subscriptionsToRemove.length; i++) {
+        const subscription = subscriptionsToRemove[i];
+
+        ignoreSubscriptions[subscription.referenceId] = true;
+        setTimeout(() => delete ignoreSubscriptions[subscription.referenceId], MS_TO_IGNORE_DATA_ON_UNSUBSCRIBED);
+
+        subscription.addStateChangedCallback(onStateChanged);
+        subscription.onUnsubscribeByTagPending();
+
+        if (shouldDisposeSubscription) {
+            subscription.dispose();

```

Comment:
Disposal logic is duplicate of one found inside Streaming.prototype.disposeSubscription.
Consider extracting common part, maybe as remoteSubscription, and use here and in Streaming.prototype.disposeSubscription. Should further improve readability of unsubscribeSubscriptionByTag.